strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactControllerTest.java

```diff
@@ -125,4 +127,36 @@ public void testPackageCommonFlow(@NpmRepository(repositoryId = REPOSITORY_RELEA
                .header(HttpHeaders.CONTENT_LENGTH, equalTo(String.valueOf(Files.size(packagePath))));
     }
 
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    public void addUserTest(@NpmRepository(repositoryId = REPOSITORY_RELEASES)
+                            Repository repository)
+    {
+        String url = getContextBaseUrl() + "/storages/{storageId}/{repositoryId}/" + NpmLayoutProvider.NPM_USER_PATH;
+
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+        NpmUser strongboxUser = new NpmUser("admin", "password");
+        NpmUser nonStrongboxUser = new NpmUser("notARealUser", "notARealPassword");
+
+        //can login with strongbox user
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+               .body(strongboxUser)
+               .when()
+               .put(url, storageId, repositoryId, strongboxUser.getName())
+               .peek()
+               .then()
+               .statusCode(HttpStatus.CREATED.value());
+
+        //can't login with non-strongbox user
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+               .body(nonStrongboxUser)
+               .when()
+               .put(url, storageId, repositoryId, nonStrongboxUser.getName())
+               .peek()
+               .then()
+               .statusCode(HttpStatus.UNAUTHORIZED.value());
+    }
```

Comment:
@LeifBrooks can you please add cases with Basic auth? I mean the one that you mentioned in chat

Reply:
Working on it right now. Could you remind me about the one I mentioned in chat?

Reply:
you tried a different cases there, and one was when you was authorized with basic auth and make add with another user

Reply:
and also it can be the case when username from url differs drom username from body

Reply:
@sbespalov I've addressed the latest concerns. Let me know if there is anything else you'd like changed. Thanks!